### PR TITLE
fix(ext/http): abort request signal when response errors

### LINKED ISSUE #10829: Native HTTP server: `Request.signal` should be aborted when request errors or is finished
The signal would be aborted on connection errors, and once the request is responded to and the body is fully sent.

Depends on #5471

### PR DESCRIPTION
**Pull request by TimoWilhelm**
Abort `Request.signal` when http response errors. Fixes #10829.

Maybe also related to #10854?

### CODE REVIEW

**Comment by bartlomieju** [comment]
Thanks for the PR @TimoWilhelm! It looks good to me implementation-wise, could you add a unit test in somewhere in `cli/tests/unit/http_test.ts` that exercises this code path and asserts that abort signal is actually fired?

**Comment by TimoWilhelm** [comment]
Thanks for the feedback @bartlomieju. I tried to create a unit test but for some reason, my changes are not picked up during the test execution... 🤔 I'm probably missing something obvious.

image

Could you maybe point me in the right direction how to setup the test environment? I didn't find anything in the [contributing docs](https://deno.land/manual@v1.31.0/references/contributing).

**Comment by bartlomieju** [comment]
Could you show me how you are running the code?

If you change something in `ext/` the easiest way to check if it works would be running `cargo test js_unit` (but it will run all unit tests), or you can do `cargo run -- test -A --unstable cli/tests/unit/http_test.ts`

**Comment by TimoWilhelm** [comment]
I was trying to run `deno test .\cli\tests\unit\http_test.ts --unstable --allow-all --filter "<my test>"`  + `--inspect-brk` for breakpoints. I'll give it a try using `cargo`.

**Comment by bartlomieju** [comment]
> I was trying to run `deno test .\cli\tests\unit\http_test.ts --unstable --allow-all --filter "<my test>"` + `--inspect-brk` for breakpoints. I'll give it a try using `cargo`.

That won't work as you will be using an already existing version of Deno - to fix the issue you will need to build Deno from source and that will pull in your changes into the Deno binary. Let me know if you hit any snags.

**Comment by TimoWilhelm** [comment]
Okay, thanks for the help worked like a charm 👍 I added a test. Not sure about the best way to wrap the `createRespondWith`. Let me know if you prefer inline other a different naming.

File: ext/http/01_http.js [hunk 1]
```diff
@@ -135,22 +135,28 @@ class HttpConn {
...
+    const responseHandler = createRespondWith(
       this,
       streamRid,
       request,
       this.#remoteAddr,
       this.#localAddr,
     );
 
+    const respondWith = (resp) =>
+      responseHandler(resp).catch((error) => {
```
Comment (bartlomieju) [inline ext/http/01_http.js:155]: This bit should use a primordial - instead of `.catch()` it should be `PromisePrototypeCatch`

Comment (bartlomieju) [reply ext/http/01_http.js:155]: Actually I think you could just pass `abortController` to `createRespondWith` and add a `catch` block there

Comment (TimoWilhelm) [reply ext/http/01_http.js]: Good idea, that's a lot cleaner 👍

**Review by aapoalas — APPROVED** [review]
LGTM. Thank you for the contribution, really neat solution!

**Pull request merged**